### ISSUE TITLE
MEL DeleteNode error

### ISSUE THREAD

**Issue by sandeepkothinti** [Microsoft/CNTK#792]
Hi, I am trying modify my network using MEL. I have a node CE which is CrossEntropy with SoftMax over a logit. I am trying to delete that node so that i can redefine the network. I am able to delete CE.SM which is my softmax node, but when i try to delete other nodes. i get this error

```
[CALL STACK]
[0x5e3ac8]          Microsoft::MSR::CNTK::ComputationNode<float>::  SetInput  (unsigned long,  std::shared_ptr<Microsoft::MSR::CNTK::ComputationNodeBase> const&) + 0x1f8
[0x812fc1]          Microsoft::MSR::CNTK::ComputationNetwork::  DeleteNode  (std::basic_string<wchar_t,std::char_traits<wchar_t>,std::allocator<wchar_t>> const&) + 0x121
[0x62bfdf]          Microsoft::MSR::CNTK::MELScript<float>::  CallFunction  (std::string const&,  Microsoft::MSR::CNTK::ConfigParamList const&) + 0x2c1f
[0x63061a]          Microsoft::MSR::CNTK::MELScript<float>::  ParseValue  (std::string const&,  unsigned long,  unsigned long) + 0x4fa
[0x5b6334]          Microsoft::MSR::CNTK::ConfigParser::  Parse  (std::string const&,  unsigned long) + 0x254
[0x623131]          Microsoft::MSR::CNTK::MELScript<float>::  FileParse  (std::string const&) + 0x341
[0x2b126fba6584]    Microsoft::MSR::CNTK::ConfigParser::  LoadConfigFileAndResolveVariables  (std::basic_string<wchar_t,std::char_traits<wchar_t>,std::allocator<wchar_t>> const&,  Microsoft::MSR::CNTK::ConfigParameters const&) + 0x64
[0x641465]          void  DoEdit  <float>(Microsoft::MSR::CNTK::ConfigParameters const&) + 0x3a5
[0x60b9fc]          void  DoCommands  <float>(Microsoft::MSR::CNTK::ConfigParameters const&,  std::shared_ptr<Microsoft::MSR::CNTK::MPIWrapper> const&) + 0xccc
[0x5ab9c7]          wmainOldCNTKConfig  (int,  wchar_t**)              + 0xbd7
[0x5ac097]          wmain1  (int,  wchar_t**)                          + 0x77
[0x5a0498]          main                                               + 0xd8
[0x2b1272d2ab15]    __libc_start_main                                  + 0xf5
[0x5a4215]
EXCEPTION occurred: an ComputationNodeBasePtr of mismatching precision was passed

```

This is my MEL script

```
m1 = LoadModel("$currModel$", format="cntk")
SetDefaultModel(m1)
HDim =2048
labelDim=$labelDim$
L$newLayer$ = SBFF(L$currLayer$.S, HDim, HDim)
DeleteNode(CE.SM)
#DeleteNode(CE.BFF)
DeleteNode(CE.T)
#DeleteNode(CE.B)
#DeleteNode(CE.W)
DeleteNode(Err)
CE = SMBFF1(L$newLayer$,labelDim,HDim,labels,tag=Criteria)
logPrior = LogPrior(labels)
ScaledLogLikelihood=Minus(LogSoftmax(CE.BFF),logPrior,tag=Output)
Err=ErrorPrediction(labels, CE.BFF, tag=Eval)
SetInput(CE.T, 1, L$newLayer$.S)
SetInput(CE.SM, 0, labels)
SetInput(L$newLayer$.*.T, 1, L$currLayer$.S)
SaveModel(m1, "$newModel$", format="cntk")
Dump(m1, "$newModel$.dump.txt")
```

 I am not sure if i am missing something. But i have followed structure mentioned in MEL documentation. Please let me know what exactly is the error.

**Comment by sandeepkothinti** [comment]
@frankseide Could you please look into this? 
I also wanted to ask if it is possible to pretrain with denoising autoencoders using dropout components. If so what kind of dropout rates, learning rates i should use? I will be using these for speech recognition.

**Comment by zhangyike** [comment]
I have the same problem with you. I wonder if you have solved it. And could you please give me any suggestions?

**Comment by sandeepkothinti** [comment]
Hi, This is something to do with order of the nodes. If you want to delete any nodes, you have to delete them in an ordered manner. So try to visualize the graph and delete all higher level dependent nodes before deleting the actual node. Hope that helps

**Comment by zhangyike** [comment]
@sandeepkothinti Thanks for your advice, I will try it.

**Comment by cha-zhang** [comment]
Not sure if this has been resolved. I'm closing it since it's old and there are no recent updates. Feel free to open a new issue if there are still problems.

**Comment by xiezhq-hermann** [comment]
https://github.com/Microsoft/CNTK/issues/3174
Could you please refer this issue and give me some advices?
Thanks a lot!
